kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -116,9 +116,12 @@ private void processKieService(ClassLoader classLoader, String key, String value
                     childServices.computeIfAbsent( serviceName, k -> new ArrayList<>() )
                             .add( newInstance( classLoader, value.substring( 1 ) ) );
                 } else {
-                    Object service = newInstance( classLoader, value );
-                    ServicePriority servicePriority = service.getClass().getAnnotation( ServicePriority.class );
-                    int priority = servicePriority != null ? servicePriority.value() : 0;
+                    int priority = 0;
+                    int separatorPos = value.indexOf( ';' );
+                    if (separatorPos > 0) {
+                        priority = Integer.parseInt( value.substring( separatorPos+1 ).trim() );
```

Comment:
@mariofusco 
This may throw a couple of exception: NPE and NumberFormat exception. Maybe using 
public String[] split(String regex) 
remove one of them...

Reply:
@gitgabrio I added one more check to be sure that the ';' is not the last char of the string and in that case I just ignore it. Regarding the NFE it's ok for me to let it be thrown because it means that user put in an invalid value for priority and there isn't any form of recovery that I could do from that error.